assets/scripts/datalayer/common.js

```diff
@@ -153,6 +159,12 @@ document.shopAnalytics = {
       $this.data('list', current_product_list_type);
     });
 
+    // Discards products in the main products list that are already tracked.
+    if ($products.length > products_list_count) {
+      $products = $products.filter('body.woocommerce ul.products .shop-analytics-product-details').slice(products_list_count);
+      products_list_count = $products.length;
```

Comment:
This assumes that further products are lazy-loaded on the same page, which is not necessarily true, as the product listing may also use a simple pager.  It also does not really prevent duplicates, as the individual product IDs are not actually recorded and diffed.

A safe approach would be to record the last seen DOM nodes (with or without jQuery object) and then use https://api.jquery.com/not/ to extract the new ones.  This still makes an assumption that in case of additional lazy-loaded products the DOM nodes of previously existing products are still the same and were not replaced with new ones.  The most safe approach would be to record the exact product IDs and filter away all seen ones, of course.

Reply:
> The most safe approach would be to record the exact product IDs and filter away all seen ones, of course.

I will try to implement something on this line as a future improvement, thanks.